internal/cluster/workflow/health_check_activity.go

```diff
@@ -0,0 +1,59 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflow
+
+import (
+	"context"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+const HealthCheckActivityName = "k8s-health-check"
+
+type HealthCheckActivityInput struct {
+	SecretID string
+}
+
+type HealthCheckActivity struct {
+	checker       HealthChecker
+	clientFactory ClientFactory
+}
+
+// +testify:mock:testOnly=true
+
+// HealthChecker returns the result of the healthcheck
+type HealthChecker interface {
+	Check(ctx context.Context, client kubernetes.Interface) error
+}
```

Comment:
**Question**: are we sure this is the best place for this interface?
(Feels unusual to me to put an interface into an activity, but I can be convinced it is only going to be used from here, but then I would ask why is it an interface if only one implementation is ever going to exist or shouldn't it be inlined instead, so we can discuss this.)

Reply:
In this way, I can easily mock it in tests.

Reply:
Let;s keep it here then and we will move it if a better place emerges.